### PR TITLE
Registration

### PR DESCRIPTION
**Pull request by kuzmavictor**
This project implements registration form that was created on ES6. This form contains fields:
- for loading image(format: `jpeg` and less than `500kB`);
- for inputting of login;
- for inputting of password ( you must confirm password in next field).

At this pahse, this project does not have enough documentation and it requires to refactoring.

### CODE REVIEW

**Comment by kuzmavictor** [comment]
@dkalinin PTAL.

File: app/service/validation/index.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+const typeImage = 'image/png';
+
+/**
+ * This  class  for validation data that was inputted from registration form.
+ */
+export class ValidationService {
+    _errorDTO;
+
+    constructor(errorDTO) {
+        this._errorDTO = errorDTO;
```
Comment (dkalinin) [inline app/service/validation/index.js]: Please refactor to not use _errorDTO as a property and state holder of a ValidationService instance. Validation service is stateless, it should receive some data to validate and return validation errors.

File: app/view/component/button/index.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ *This class allows to render new {@link HTML Element} of button
+ */
+export class Button extends Component {
+    /**
+     * The constructor init creating component of button.
+     * @param parameters - the parameters for creating button  element.
+     */
+    constructor(parameters) {
+        super(parameters.id);
+        this.root = parameters.parentRoot;
```
Comment (dkalinin) [inline app/view/component/button/index.js]: Let's not include parentRoot to parameters, make it a separate parameter.

File: app/view/component/button/index.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import {Component} from "../index";
+
+/**
+ *This class allows to render new {@link HTML Element} of button
+ */
+export class Button extends Component {
+    /**
+     * The constructor init creating component of button.
+     * @param parameters - the parameters for creating button  element.
```
Comment (dkalinin) [inline app/view/component/button/index.js]: Type? Keys? Values?

File: app/view/component/form-field/index.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+ * This class is the class for field form. It is custom element that contains other
+ * elements.It contains input and two labels element.
+ *
+ */
+export class FormField extends Component {
+    _textError = "";
+
+    /**
+     * The constructor creates {@link FormField} component for a form.
+     * @param {Object} parameters - the parameters for creating
```
Comment (dkalinin) [inline app/view/component/form-field/index.js]: What keys should I pass?

File: app/view/component/form-field/index.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    /**
+     * Render element {@link Label} and {@link Input} for this componen{@link FormField}.
+     * @private
+     */
+    _render() {
+        this.htmlDivElement = document.createElement(DIV_TAG);
+        this.label = new Label(this._labelId, this.htmlDivElement, "black");
+        this.label.text = this._textLabel;
+        this.input = new Input(this._componentId, this.htmlDivElement, this._placeHolder, this._inputType);
+        this.errorLabel = new Label(this._errorLableId, this.htmlDivElement, "red");
```
Comment (dkalinin) [inline app/view/component/form-field/index.js]: Label tag is used for inputs only. For text use a proper tag like span or div.

File: app/view/component/form-field/index.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+     */
+    reRender() {
+        this.errorLabel.reRender();
+        this.errorLabel.text = this._textError;
+    }
+
+    /**
+     * Returns data from input field(file or value).
+     *
+     * @param parameter - type of value (`file` or `value`)
```
Comment (dkalinin) [inline app/view/component/form-field/index.js]: It should be a part of implementation. Use _inputType to find out which attribute to get.

File: app/view/component/form-field/index.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    /**
+     * Add task  for render this component use {@link EventBus}.
+     * @param eventName
+     */
+    addRenderTask(eventName) {
+        this._eventBus.subscribe(eventName, () => {
+            this._render();
+        });
```
Comment (dkalinin) [inline app/view/component/form-field/index.js]: Event bus is used in incorrect way. Please remove all code related to event bus and refactor your project not to use event bus.

File: app/view/component/form-field/index.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+        this._placeHolder = parameters.placeholder;
+        this._inputType = parameters.inputType;
+        this._root = parameters.parentRoot;
+        this._eventBus = parameters.eventBus;
+    }
+
+    /**
+     * Sets fields {#_textError} (the error message that/
+     *
+     * will be show if inputting data is incorrect).
```
Comment (dkalinin) [inline app/view/component/form-field/index.js]: Pay attention on formatting and correct English in JS doc. Use Grammarly or any other tool to check your documentation.

**Pull request merged**